src/operations.rs

```diff
@@ -121,28 +125,52 @@ impl SwapOperationsListUnchecked {
         }
 
         let mut prev_ask_asset = operations.first().unwrap().ask_asset_info.clone();
+        visited_assets.insert(operations.first().unwrap().offer_asset_info.to_string());
+        visited_assets.insert(prev_ask_asset.to_string());
+        assets_to_swap.insert(prev_ask_asset.to_string());
         for operation in operations.iter().skip(1) {
-            if operation.offer_asset_info != prev_ask_asset {
+            if operation.offer_asset_info != prev_ask_asset
+                || visited_assets.contains(&operation.ask_asset_info.to_string())
+            {
                 return Err(ContractError::InvalidSwapOperations { operations });
             }
             prev_ask_asset = operation.ask_asset_info.clone();
+            assets_to_swap.remove(&operation.offer_asset_info.to_string());
+            assets_to_swap.insert(operation.ask_asset_info.to_string());
+            visited_assets.insert(operation.ask_asset_info.to_string());
```

Comment:
Could we add a `simulate_swap` query for each pool to validate that the pool actually exists? Maybe we should do that in the `Pool::new()` implementation? @apolloshab  @apollo-sturdy